modules/calcite/src/test/java/org/apache/ignite/internal/processors/query/calcite/jdbc/JdbcQueryTest.java

```diff
@@ -108,6 +112,65 @@ public void testSimpleQuery() throws Exception {
         stmt.close();
     }
 
+    /** Test batched execution of statement. */
+    @Test
+    public void testBatch() throws Exception {
+        stmt.execute("CREATE TABLE Person(\"id\" INT, PRIMARY KEY(\"id\"), \"name\" VARCHAR)");
+
+        for (int i = 0; i < 10; ++i)
+            stmt.addBatch(String.format("INSERT INTO Person VALUES (%d, 'Name')", i));
+
+        stmt.executeBatch();
+
+        try (ResultSet rs = stmt.executeQuery("select * from Person p order by \"id\" asc")) {
+            for (int i = 0; rs.next(); ++i) {
+                assertEquals(i, rs.getInt(1));
+                assertEquals("Name", rs.getString(2));
+            }
+        }
+    }
+
+    /** Test batched execution of prepared statement. */
+    @Test
+    public void testBatchPrepared() throws Exception {
+        stmt.execute("CREATE TABLE Person(\"id\" INT, PRIMARY KEY(\"id\"), \"name\" VARCHAR)");
+
+        try (PreparedStatement stmt0 = conn.prepareStatement("INSERT INTO Person VALUES (?, ?);" +
+            "INSERT INTO Person VALUES (?, ?)")) {
+            stmt0.setInt(1, 0);
+            stmt0.setString(2, "Name");
+            stmt0.setInt(2, 1);
+            stmt0.setString(4, "Name");
+            stmt0.addBatch();
+
+            GridTestUtils.assertThrows(log, stmt0::executeBatch, BatchUpdateException.class,
+                "Multiline statements are not supported in batched query");
+        }
+
+        int[] ret;
+        try (PreparedStatement stmt0 = conn.prepareStatement("INSERT INTO Person VALUES (?, ?), (?, ?)")) {
+            for (int i = 0; i < 1000; i += 2) {
+                stmt0.setInt(1, i);
+                stmt0.setString(2, "Name");
+                stmt0.setInt(3, i + 1);
+                stmt0.setString(4, "Name");
+                stmt0.addBatch();
+            }
+
+            ret = stmt0.executeBatch();
+        }
+
+        try (ResultSet rs = stmt.executeQuery("select * from Person p order by \"id\" asc")) {
+            int i = 0;
+            for (; rs.next(); ++i) {
+                assertEquals(i, rs.getInt(1));
+                assertEquals("Name", rs.getString(2));
+            }
+            assertEquals(ret.length * 2, i);
+            assertTrue(Arrays.stream(ret).anyMatch(k -> k == 2));
+        }
+    }
```

Comment:
Drop table and close statement should be moved to finally block, but was totally removed

Reply:
It is not needed since stmt closed and statement stoped in afterTest